### PR TITLE
[Chef] Fix Basic Video Player broken

### PR DESCRIPTION
**Pull request by erwinpan1**
Chef video player was broken due to #30691 which upgraded class interfaces in src/app/clusters/channel-server/ . 
Now I migrate the codes in examples/tv-app/ to rebuild new BasicVideoPlayer for Chef

### CODE REVIEW

File: examples/chef/common/clusters/audio-output/AudioOutputManager.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+using namespace std;
+using namespace chip::app;
+using namespace chip::app::Clusters::AudioOutput;
+
+AudioOutputManager::AudioOutputManager()
+{
+    mCurrentOutput = 1;
+
+    for (int i = 1; i < 4; ++i)
```
Comment (andy31415) [inline examples/chef/common/clusters/audio-output/AudioOutputManager.cpp:31]: why is 4 a magic number here? can we add a constant for it?

File: examples/chef/common/clusters/audio-output/AudioOutputManager.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        OutputInfoType outputInfo;
+        outputInfo.outputType = chip::app::Clusters::AudioOutput::OutputTypeEnum::kHdmi;
+        // note: safe only because of use of string literal
+        outputInfo.name  = chip::CharSpan::fromCharString("HDMI");
+        outputInfo.index = static_cast<uint8_t>(i);
```
Comment (andy31415) [inline examples/chef/common/clusters/audio-output/AudioOutputManager.cpp:37]: why not make i a `uint8_t` to start with ?

File: examples/chef/common/clusters/audio-output/AudioOutputManager.cpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            ReturnErrorOnFailure(encoder.Encode(outputInfo));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+bool AudioOutputManager::HandleRenameOutput(const uint8_t & index, const chip::CharSpan & name)
+{
+    // TODO: Insert code here
```
Comment (andy31415) [inline examples/chef/common/clusters/audio-output/AudioOutputManager.cpp:61]: same: do we need to clearup some todos?

File: examples/chef/common/clusters/audio-output/AudioOutputManager.cpp [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    bool audioOutputRenamed = false;
+
+    for (OutputInfoType & output : mOutputs)
+    {
+        if (output.index == index)
+        {
+            audioOutputRenamed = true;
+            memcpy(this->Data(index), name.data(), name.size());
```
Comment (andy31415) [inline examples/chef/common/clusters/audio-output/AudioOutputManager.cpp:69]: how can we ensure that there is enough storage space in Data to store the name?

File: examples/chef/common/clusters/audio-output/AudioOutputManager.cpp [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+bool AudioOutputManager::HandleSelectOutput(const uint8_t & index)
+{
+    // TODO: Insert code here
+    bool audioOutputSelected = false;
+    for (OutputInfoType & output : mOutputs)
+    {
+        if (output.index == index)
+        {
+            audioOutputSelected = true;
```
Comment (andy31415) [inline examples/chef/common/clusters/audio-output/AudioOutputManager.cpp:85]: do we expect multiple outputs to have the same index?

Otherwise both here and in the other looping, you can return right away `return true` and not have this extra variable of `audioOutputSelected` or `audioOutputRenamed`.

You may even have a function of FindOutputWithIndex(index) to be shared among the functions.

File: examples/chef/common/clusters/audio-output/AudioOutputManager.h [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/audio-output-server/audio-output-server.h>
+#include <vector>
+
+using chip::app::AttributeValueEncoder;
```
Comment (andy31415) [inline examples/chef/common/clusters/audio-output/AudioOutputManager.h:24]: using is not allowed in headers at the top level. Please remove.

File: examples/chef/common/clusters/audio-output/AudioOutputManager.h [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    uint8_t HandleGetCurrentOutput() override;
+    CHIP_ERROR HandleGetOutputList(AttributeValueEncoder & aEncoder) override;
+    bool HandleRenameOutput(const uint8_t & index, const chip::CharSpan & name) override;
+    bool HandleSelectOutput(const uint8_t & index) override;
+    char * Data(uint8_t index) { return mCharDataBuffer[index]; }
+
+protected:
+    uint8_t mCurrentOutput;
+    std::vector<OutputInfoType> mOutputs;
+    // Magic numbers are here on purpose, please allocate memory
```
Comment (andy31415) [inline examples/chef/common/clusters/audio-output/AudioOutputManager.h:42]: have constants for sizes, so you can refer to them in the cpp file for bounds checks.

File: examples/chef/common/clusters/channel/ChannelManager.cpp [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+    program3.subtitle   = MakeOptional(chip::CharSpan::fromCharString("My Program Subtitle3"));
+    program3.startTime  = 0;
+    program3.endTime    = 60 * 60;
+
+    mPrograms.push_back(program3);
+}
+
+CHIP_ERROR ChannelManager::HandleGetChannelList(AttributeValueEncoder & aEncoder)
+{
+    // TODO: Insert code here
```
Comment (andy31415) [inline examples/chef/common/clusters/channel/ChannelManager.cpp:111]: please remove more of these "insert code here" bits

File: examples/chef/common/clusters/channel/ChannelManager.cpp [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+
+CHIP_ERROR ChannelManager::HandleGetCurrentChannel(AttributeValueEncoder & aEncoder)
+{
+    return aEncoder.Encode(mCurrentChannel);
+}
+
+bool isChannelMatched(const ChannelInfoType & channel, const CharSpan & match)
+{
+    char number[32];
+    sprintf(number, "%d.%d", channel.majorNumber, channel.minorNumber);
```
Comment (andy31415) [inline examples/chef/common/clusters/channel/ChannelManager.cpp:140]: Prefer usage of StringBuilder to build strings instead of sprintf and static chars. That one guarantees null termination and no overflows.

File: examples/chef/common/clusters/channel/ChannelManager.cpp [hunk 10]
```diff
@@ -0,0 +1,343 @@
...
+    bool nameMatch = channel.name.HasValue() ? channel.name.Value().data_equal(match) : false;
```
Comment (andy31415) [inline examples/chef/common/clusters/channel/ChannelManager.cpp:141]: instead of `foo ? bar : false` you can use `foo && bar`

Comment (andy31415) [reply examples/chef/common/clusters/channel/ChannelManager.cpp:141]: This whole thing could be:

```cpp
auto isMatch = [&match](const Optional<chip::CharSpan> &a)
{
    return a.HasValue() && a.Value().data_equal(match);
};

return isMatch(channel.name) 
    || isMatch(channel.affiliateCallSign) 
    || isMatch(channel.callSign)
    || match.data_equal(chip::CharSpan::fromCharString(number));
 ```
 
 not sure if more readable, howver it seems to have less complexity.

File: examples/chef/common/clusters/channel/ChannelManager.cpp [hunk 11]
```diff
@@ -0,0 +1,343 @@
...
+    uint16_t index = 0;
+    for (auto const & channel : mChannels)
+    {
+        // verify if CharSpan matches channel name
+        // or callSign or affiliateCallSign or majorNumber.minorNumber
+        if (isChannelMatched(channel, match))
+        {
+            matchedChannels.push_back(channel);
+        }
+        else if (matchedChannels.size() == 0)
```
Comment (andy31415) [inline examples/chef/common/clusters/channel/ChannelManager.cpp:162]: use `std::vector::empty` for clearer code.

File: examples/chef/common/clusters/channel/ChannelManager.cpp [hunk 12]
```diff
@@ -0,0 +1,343 @@
...
+        {
+            // "index" is only used when we end up with matchedChannels.size() == 1.
+            // In that case, we want it to be the number of non-matching channels we saw before
+            // the matching one.
+            index++;
```
Comment (andy31415) [inline examples/chef/common/clusters/channel/ChannelManager.cpp:167]: rename index to something explaining usage. I think something like `channelsBeforeFirstMatch` or similar.

File: examples/chef/common/clusters/channel/ChannelManager.cpp [hunk 13]
```diff
@@ -0,0 +1,343 @@
...
+        }
+    }
+
+    ChangeChannelResponseType response;
+
+    // Error: Found multiple matches
+    if (matchedChannels.size() > 1)
```
Comment (andy31415) [inline examples/chef/common/clusters/channel/ChannelManager.cpp:174]: why do we go through all channels to find this? If this is a critical error it could be done during the loop: if we match and we already had a match (vector is non-empty) then return an error.

Further more, we do not need a vector anymore at all - we only need to store one match.

File: examples/chef/common/clusters/channel/ChannelManager.cpp [hunk 14]
```diff
@@ -0,0 +1,343 @@
...
+    {
+        response.status = chip::app::Clusters::Channel::StatusEnum::kMultipleMatches;
+        helper.Success(response);
+    }
+    else if (matchedChannels.size() == 0)
```
Comment (andy31415) [inline examples/chef/common/clusters/channel/ChannelManager.cpp:179]: use empty.

Also return after this and drop the else, to reduce complexity.

File: examples/chef/common/clusters/channel/ChannelManager.cpp [hunk 15]
```diff
@@ -0,0 +1,343 @@
...
+        if (channel.minorNumber == minorNumber && channel.majorNumber == majorNumber)
+        {
+            // verify if channel changed by comparing values of current channel with the requested channel
+            if (channel.minorNumber != mCurrentChannel.minorNumber || channel.majorNumber != mCurrentChannel.majorNumber)
+            {
+                channelChanged       = true;
+                mCurrentChannelIndex = index;
+                mCurrentChannel      = channel;
+            }
+        }
```
Comment (andy31415) [inline examples/chef/common/clusters/channel/ChannelManager.cpp:211]: you can early-return true right away here (or do we expect multiple matches?)

File: examples/chef/common/clusters/channel/ChannelManager.cpp [hunk 16]
```diff
@@ -0,0 +1,343 @@
...
+    // channelPaging.nextToken = MakeOptional<DataModel::Nullable<Structs::PageTokenStruct::Type>>(paging);
+
+    std::vector<ProgramType> matches;
+    for (auto const & program : mPrograms)
+    {
+        if (startTime.ValueOr(0) > program.startTime)
+        {
+            continue;
+        }
+        if (endTime.HasValue() && endTime.ValueOr(0) < program.endTime)
```
Comment (andy31415) [inline examples/chef/common/clusters/channel/ChannelManager.cpp:263]: since you use ValueOr, why do we need HasValue ?

File: examples/chef/common/clusters/channel/ChannelManager.cpp [hunk 17]
```diff
@@ -0,0 +1,343 @@
...
+                                         const chip::ByteSpan & data)
+{
+    // Start recording
+    std::string idString(programIdentifier.data(), programIdentifier.size());
+    for (auto & program : mPrograms)
+    {
+        std::string nextIdString(program.identifier.data(), program.identifier.size());
+        if (nextIdString == idString)
+        {
+            program.recordingFlag = MakeOptional(static_cast<uint32_t>(shouldRecordSeries ? 2 : 1));
```
Comment (andy31415) [inline examples/chef/common/clusters/channel/ChannelManager.cpp:308]: what are 2 or 1 ? please add constants for them

File: examples/chef/common/clusters/channel/ChannelManager.cpp [hunk 18]
```diff
@@ -0,0 +1,343 @@
...
+                                               const chip::ByteSpan & data)
+{
+    // Cancel recording
+    std::string idString(programIdentifier.data(), programIdentifier.size());
+    for (auto & program : mPrograms)
+    {
+        std::string nextIdString(program.identifier.data(), program.identifier.size());
+        if (nextIdString == idString)
+        {
+            program.recordingFlag = MakeOptional(static_cast<uint32_t>(0));
```
Comment (andy31415) [inline examples/chef/common/clusters/channel/ChannelManager.cpp:326]: does `MakeOptional<uint32_t>(0)` work to make things shorter?

File: examples/chef/common/clusters/channel/ChannelManager.h [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+using RecordingFlagBitmap       = chip::app::Clusters::Channel::RecordingFlagBitmap;
+using ChannelDelegate           = chip::app::Clusters::Channel::Delegate;
+using ChangeChannelResponseType = chip::app::Clusters::Channel::Commands::ChangeChannelResponse::Type;
+using ProgramGuideResponseType  = chip::app::Clusters::Channel::Commands::ProgramGuideResponse::Type;
+using ChannelInfoType           = chip::app::Clusters::Channel::Structs::ChannelInfoStruct::Type;
+using AdditionalInfoType        = chip::app::Clusters::Channel::Structs::AdditionalInfoStruct::Type;
+using LineupInfoType            = chip::app::Clusters::Channel::Structs::LineupInfoStruct::Type;
+using PageTokenType             = chip::app::Clusters::Channel::Structs::PageTokenStruct::Type;
+using ProgramType               = chip::app::Clusters::Channel::Structs::ProgramStruct::Type;
+using ChannelPagingType         = chip::app::Clusters::Channel::Structs::ChannelPagingStruct::Type;
```
Comment (andy31415) [inline examples/chef/common/clusters/channel/ChannelManager.h:35]: no using in headers please.

File: examples/chef/common/clusters/channel/ChannelManager.h [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+    uint32_t GetFeatureMap(chip::EndpointId endpoint) override;
+
+protected:
+    uint16_t mCurrentChannelIndex;
+    ChannelInfoType mCurrentChannel;
+    std::vector<ChannelInfoType> mChannels;
+    std::vector<ProgramType> mPrograms;
+
+private:
+    // TODO: set this based upon meta data from app
```
Comment (andy31415) [inline examples/chef/common/clusters/channel/ChannelManager.h:74]: why TODO here? could we have this as a constexpr instead?

File: examples/chef/common/clusters/keypad-input/KeypadInputManager.cpp [hunk 21]
```diff
@@ -0,0 +1,113 @@
...
+#include "KeypadInputManager.h"
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/util/config.h>
+
+using namespace chip;
+using namespace chip::app::Clusters::KeypadInput;
+
+void KeypadInputManager::HandleSendKey(CommandResponseHelper<SendKeyResponseType> & helper, const CecKeyCodeType & keycCode)
+{
+    // TODO: Insert code here
```
Comment (andy31415) [inline examples/chef/common/clusters/keypad-input/KeypadInputManager.cpp:30]: please remove the TODOs if done, otherwise explain what code is needed.

File: examples/chef/common/clusters/keypad-input/KeypadInputManager.h [hunk 22]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+#pragma once
+
+#include <app/clusters/keypad-input-server/keypad-input-server.h>
+
+using chip::app::CommandResponseHelper;
+using KeypadInputDelegate = chip::app::Clusters::KeypadInput::Delegate;
+using SendKeyResponseType = chip::app::Clusters::KeypadInput::Commands::SendKeyResponse::Type;
+using CecKeyCodeType      = chip::app::Clusters::KeypadInput::CECKeyCodeEnum;
```
Comment (andy31415) [inline examples/chef/common/clusters/keypad-input/KeypadInputManager.h:26]: no using in headers.
Comments from other headers apply: why is the featuremap a member with a TODO on it?

File: examples/chef/common/clusters/media-input/MediaInputManager.cpp [hunk 23]
```diff
@@ -0,0 +1,111 @@
...
+
+using namespace std;
+using namespace chip;
+using namespace chip::app::Clusters::MediaInput;
+
+MediaInputManager::MediaInputManager()
+{
+    mCurrentInput = 1;
+
+    for (int i = 1; i < 3; ++i)
```
Comment (andy31415) [inline examples/chef/common/clusters/media-input/MediaInputManager.cpp:30]: Comments from other CPP apply: add constants for magic numbers, remove TODO:insert or explain what code needs to be added. This applies throughout the PR.

File: examples/chef/common/clusters/media-input/MediaInputManager.h [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+public:
+    MediaInputManager();
+
+    CHIP_ERROR HandleGetInputList(chip::app::AttributeValueEncoder & aEncoder) override;
+    uint8_t HandleGetCurrentInput() override;
+    bool HandleSelectInput(const uint8_t index) override;
+    bool HandleShowInputStatus() override;
+    bool HandleHideInputStatus() override;
+    bool HandleRenameInput(const uint8_t index, const chip::CharSpan & name) override;
+    char * Data(uint8_t index) { return mCharDataBuffer[index]; }
```
Comment (andy31415) [inline examples/chef/common/clusters/media-input/MediaInputManager.h:38]: similar comments here: return MutableByteSpan, check ranges.

File: examples/chef/common/stubs.cpp [hunk 25]
```diff
@@ -340,14 +342,132 @@ bool emberAfPluginDoorLockSetCredential(chip::EndpointId endpointId, uint16_t cr
...
+    }
+}
+
+/** @brief OnOff Cluster Init
+ *
+ * This function is called when a specific cluster is initialized. It gives the
+ * application an opportunity to take care of cluster initialization procedures.
+ * It is called exactly once for each endpoint where cluster is present.
+ *
+ * @param endpoint   Ver.: always
```
Comment (andy31415) [inline examples/chef/common/stubs.cpp:374]: what is this comment? What is Ver. always?

File: examples/chef/common/clusters/audio-output/AudioOutputManager.cpp [hunk 26]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+uint8_t AudioOutputManager::HandleGetCurrentOutput()
+{
+    return mCurrentOutput;
+}
+
+CHIP_ERROR AudioOutputManager::HandleGetOutputList(AttributeValueEncoder & aEncoder)
+{
+    // TODO: Insert code here
```
[the remark this answers is not in the document]
Comment (erwinpan1) [reply examples/chef/common/clusters/audio-output/AudioOutputManager.cpp]: Not used. We can remove them.

File: examples/chef/common/clusters/audio-output/AudioOutputManager.cpp [hunk 27]
```diff
@@ -0,0 +1,92 @@
...
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/util/config.h>
+#ifdef EMBER_AF_PLUGIN_AUDIO_OUTPUT_SERVER
```
[the remark this answers is not in the document]
Comment (erwinpan1) [reply examples/chef/common/clusters/audio-output/AudioOutputManager.cpp:20]: @andy31415 Clusters could be disabled when building non-related devices. Can you share your experiences on how to do this in CMakeLists.txt and gn file ?

File: examples/chef/common/clusters/audio-output/AudioOutputManager.cpp (hunk 2 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/audio-output/AudioOutputManager.cpp:37]: Done.

File: examples/chef/common/clusters/audio-output/AudioOutputManager.cpp (hunk 3 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/audio-output/AudioOutputManager.cpp]: Removal done. Thanks.

File: examples/chef/common/clusters/audio-output/AudioOutputManager.cpp (hunk 4 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/audio-output/AudioOutputManager.cpp]: Fix done. Thanks for figuring out this potential memory violation.

File: examples/chef/common/clusters/audio-output/AudioOutputManager.h (hunk 6 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/audio-output/AudioOutputManager.h:24]: Fix done. Thanks.

File: examples/chef/common/clusters/audio-output/AudioOutputManager.h (hunk 7 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/audio-output/AudioOutputManager.h:42]: @andy31415 I think we can still mOutputs.size() for bounds check.  On the other hand, not defining as const could provide a flexibility e.g. Modifying the mOutput through RPC in case we need that.

File: examples/chef/common/clusters/audio-output/AudioOutputManager.cpp (hunk 5 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/audio-output/AudioOutputManager.cpp:85]: done.

File: examples/chef/common/clusters/channel/ChannelManager.cpp (hunk 8 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.cpp]: done.

File: examples/chef/common/clusters/channel/ChannelManager.cpp (hunk 9 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.cpp]: done

File: examples/chef/common/clusters/channel/ChannelManager.cpp (hunk 10 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.cpp]: It seems good. Done.

File: examples/chef/common/clusters/audio-output/AudioOutputManager.h [hunk 28]
```diff
@@ -0,0 +1,44 @@
...
+class AudioOutputManager : public AudioOutputDelegate
+{
+public:
+    AudioOutputManager();
+
+    uint8_t HandleGetCurrentOutput() override;
+    CHIP_ERROR HandleGetOutputList(AttributeValueEncoder & aEncoder) override;
+    bool HandleRenameOutput(const uint8_t & index, const chip::CharSpan & name) override;
+    bool HandleSelectOutput(const uint8_t & index) override;
+    char * Data(uint8_t index) { return mCharDataBuffer[index]; }
```
[the remark this answers is not in the document]
Comment (erwinpan1) [reply examples/chef/common/clusters/audio-output/AudioOutputManager.h:34]: > Is this needed externally? Why public? 

This is inherited from parent [delegate](https://github.com/project-chip/connectedhomeip/blob/master/src/app/clusters/audio-output-server/audio-output-delegate.h#L39-L41)

File: examples/chef/common/clusters/channel/ChannelManager.cpp (hunk 12 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.cpp:169]: OK. Rename it to `iFirstMatchedChannel`

File: examples/chef/common/clusters/channel/ChannelManager.cpp (hunk 8 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.cpp]: done.

File: examples/chef/common/clusters/channel/ChannelManager.cpp (hunk 9 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.cpp]: done

File: examples/chef/common/clusters/channel/ChannelManager.cpp (hunk 11 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.cpp]: `matchedChannels` removed

File: examples/chef/common/clusters/channel/ChannelManager.cpp (hunk 13 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.cpp:173]: matchedChannels removed.

File: examples/chef/common/clusters/channel/ChannelManager.cpp (hunk 14 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.cpp]: `matchedChannels` removed

File: examples/chef/common/clusters/channel/ChannelManager.cpp (hunk 15 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.cpp:206]: done

File: examples/chef/common/clusters/channel/ChannelManager.cpp (hunk 16 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.cpp:258]: we don't need `endTime.HasValue()`

File: examples/chef/common/clusters/channel/ChannelManager.cpp (hunk 17 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.cpp:303]: Modified to 
```
program.recordingFlag = MakeOptional(static_cast<uint32_t>(shouldRecordSeries ? RecordingFlagBitmap::kRecordSeries : RecordingFlagBitmap::kScheduled));
```

File: examples/chef/common/clusters/channel/ChannelManager.cpp (hunk 18 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.cpp:321]: yes. It works. Thanks.

File: examples/chef/common/clusters/channel/ChannelManager.h (hunk 19 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/channel/ChannelManager.h:34]: Removed.

File: examples/chef/common/clusters/keypad-input/KeypadInputManager.h (hunk 22 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/keypad-input/KeypadInputManager.h:26]: Removed using from top scope. (Just leave it in class scope)
Remove the featuremap which is use less

File: examples/chef/common/clusters/media-input/MediaInputManager.cpp (hunk 23 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/media-input/MediaInputManager.cpp]: Done.

File: examples/chef/common/stubs.cpp (hunk 25 above)
Comment (erwinpan1) [reply examples/chef/common/stubs.cpp:374]: Remoed

File: examples/chef/common/clusters/media-input/MediaInputManager.h (hunk 24 above)
Comment (erwinpan1) [reply examples/chef/common/clusters/media-input/MediaInputManager.h:37]: Done

File: .github/workflows/bloat_check.yaml [hunk 29]
```diff
@@ -33,7 +33,7 @@ jobs:
         runs-on: ubuntu-latest
 
         container:
-            image: ghcr.io/project-chip/chip-build:32
+            image: ghcr.io/project-chip/chip-build:35
```
Comment (bzbarsky-apple) [inline .github/workflows/bloat_check.yaml:36]: What do these changes have to do with the Chef video player?  It looks like this PR has a whole bunch of unrelated changes in it...

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: .github/workflows/bloat_check.yaml (hunk 29 above)
Comment (erwinpan1) [reply .github/workflows/bloat_check.yaml:36]: @bzbarsky-apple Thanks for reviewing this. I have some conflicts during rebasing so this PR contains other people's commits during rebasing. Please ignore the current one (I'll close it soon) and help to review #32091 instead.

**Comment by erwinpan1** [comment]
Close the pull request due to mixing with a lot of unrelated commit.
The new PR is created as #32091
